integration-tests/test/rnode.py

```diff
@@ -132,6 +140,24 @@ def get_rnode_address(self) -> str:
     def get_metrics(self):
         return self.shell_out('curl', '-s', 'http://localhost:40403/metrics')
 
+    def get_connected_peers_metric_value(self):
+        try:
+            return self.shell_out('sh', '-c', 'curl -s http://localhost:40403/metrics | grep ^rchain_comm_rp_connect_peers\\ ')
+        except NonZeroExitCodeError as e:
+            if e.exit_code == 1:
+                return ''
+            raise
+
+    def ensure_zero_exit_code(self) -> None:
+        try:
+            status = self.container.wait(timeout=1)
+        except requests.exceptions.ReadTimeout:
+            # Container is still running: let the test case body decide whether the test succeeded or not
+            return
```

Comment:
What happens if this branch is hit during `cleanup`?

Anyway, the function name is lying ATM: it doesn't really ensure there **is** a status code. It should either fail for unavailable status codes, or return an option-like value saying `None` (vs Some(true/false) normally) and be called `check_exit_code`. Yeah I know that's not how one does Python. I hate the fact that the callers of this function need to know its internals and have to know that this function needs to be called in a loop, as cleanup seems to do.

Change my mind.

Reply:
First of all, the `do-not-merge-yet` label should really say `do-not-review-yet`.  I added you as a reviewer because I thought that would help with bors problems.  This PR isn't fully baked yet.

Reply:
You can create such a label. Or just tell me that you used me in hopes of unclogging bors :*